plugins/module_utils/wafv2.py

```diff
@@ -0,0 +1,146 @@
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+try:
+    from botocore.exceptions import ClientError, BotoCoreError
+except ImportError:
+    pass  # caught by AnsibleAWSModule
+
+
+def wafv2_list_web_acls(wafv2, scope, fail_json_aws, Nextmarker=None):
+    # there is currently no paginator for wafv2
+    req_obj = {
+        'Scope': scope,
+        'Limit': 100
+    }
+    if Nextmarker:
+        req_obj['NextMarker'] = Nextmarker
+
+    try:
+        response = wafv2.list_web_acls(**req_obj)
+    except (BotoCoreError, ClientError) as e:
+        fail_json_aws(e, msg="Failed to list wafv2 web acl.")
+
+    if response.get('NextMarker'):
+        response['WebACLs'] += wafv2_list_web_acls(wafv2, scope, fail_json_aws, Nextmarker=response.get('NextMarker')).get('WebACLs')
+    return response
+
+
+def wafv2_list_rule_groups(wafv2, scope, fail_json_aws, Nextmarker=None):
+    # there is currently no paginator for wafv2
+    req_obj = {
+        'Scope': scope,
+        'Limit': 100
+    }
+    if Nextmarker:
+        req_obj['NextMarker'] = Nextmarker
+
+    try:
+        response = wafv2.list_rule_groups(**req_obj)
+    except (BotoCoreError, ClientError) as e:
+        fail_json_aws(e, msg="Failed to list wafv2 rule group.")
+
+    if response.get('NextMarker'):
+        response['RuleGroups'] += wafv2_list_rule_groups(wafv2, scope, fail_json_aws, Nextmarker=response.get('NextMarker')).get('RuleGroups')
+    return response
+
+
+def wafv2_snake_dict_to_camel_dict(a):
```

Comment:
May I **suggest** to add a unit-test for this function?

Reply:
Yes I know and this is the reason why I put `suggest` in bold. PLEASE, Don't force yourself to do it!

I see two reasons to write unit-tests in your case:

- sometime the behavior of the API change, it's a way to have a copy of the original expect input.
- unit-tests run way way way way faster than integration tests. And if they fail we can configure the CI to avoid the functional tests. This way we save the valuable developer time and a lot of CI resources.
- it's also rather easy for a developer to be able to run the unit-tests locally. It's a different story for the functional tests.

Reply:
For what it's worth you should be able to copy most of the unit-test framework for these transforms from https://github.com/ansible-collections/amazon.aws/blob/main/tests/unit/module_utils/core/test_scrub_none_parameters.py